src/containers/facility/FacilityIntroduction.tsx

```diff
@@ -146,42 +174,64 @@ export const FacilityIntroduction = ({
 }) => {
   const { getAccommodationById, accommodations } = useAccommodationsAndOffers();
   const { id } = useParams();
+
+  const [galleryOpen, setGalleryOpen] = useState<boolean>(false);
+  const [slideOpen, setSlideOpen] = useState<boolean>(false);
+  const [slideIndex, setSlideIndex] = useState<number>(0);
+
   const accommodation: AccommodationWithId | null = getAccommodationById(
     accommodations,
     String(id)
   );
 
-  const [galleryOpen, setGalleryOpen] = useState<boolean>(false);
-  const handleOpenGallery = () => setGalleryOpen(true);
+  const sortedImages: MediaItem[] = useMemo(
+    () => sortByLargestImage(accommodation?.media ?? []),
+    [accommodation?.media]
+  );
+
+  // get largest images and their urls
+  const largestImages = useMemo(() => getLargestImages(sortedImages), [sortedImages]);
+  const largestImagesUrls = useMemo(
+    () => largestImages.map(({ url }) => url as string),
+    [largestImages]
+  );
+
+  // slide handlers
+  const handleOpenSlide = (targetSlideIndex = 0) => {
+    setSlideIndex(targetSlideIndex);
+    setSlideOpen(true);
+  };
+
+  const handleCloseSlide = () => setSlideOpen(false);
+
+  // gallery handlers
+  const handleOpenGallery = () => {
+    if (largestImages.length > 5) {
+      setGalleryOpen(true);
+    } else {
+      handleCloseGallery();
+      handleOpenSlide();
+    }
+  };
+
   const handleCloseGallery = () => setGalleryOpen(false);
 
+  // show all photos buttons
   const buttonStyle: SxProps = {
     position: 'absolute',
     right: '2%',
     bottom: '5%'
   };
 
-  const sortedImages: MediaItem[] = useMemo(
-    () => sortByLargestImage(accommodation?.media ?? []),
-    [accommodation?.media]
-  );
-
-  const largestImages = useMemo(() => getLargestImages(sortedImages), [sortedImages]);
-
   const [mainImage, ...rest] = sortedImages;
-  const address = [
-    accommodation?.contactInformation?.address?.streetAddress,
-    accommodation?.contactInformation?.address?.locality,
-    accommodation?.contactInformation?.address?.premise,
-    accommodation?.contactInformation?.address?.country
-  ]
-    .filter(Boolean)
-    .join(', ');
+  const address = buildAccommodationAddress(accommodation);
+
+  console.log(largestImagesUrls);
```

Comment:
rm logs

Reply:
Oops! We should probably include a `console: warn` rule in EsLint?